docs/examples/creating-a-crud-api.md

```diff
@@ -0,0 +1,269 @@
+---
+description: |
+  Use HTTP CRUD methods to perform operations on resources. Learn how to use HTTP handlers to create a RESTful API.
+---
+
+The MDN [docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods) are a
+great resource to learn more about HTTP methods. We'll touch on the four
+fundamental methods necessary to create a basic CRUD (create, read, update,
+delete) API here. Additionally, we'll briefly mention CORS requests and how
+`OPTIONS` comes into play.
+
+Using HTTP methods is a common way to create a REST API. Fresh supports common
+HTTP methods in handlers out of the box. Async HTTP requests are also supported.
+Read more about custom handlers [here](/docs/getting-started/custom-handlers).
+
+In this example we'll be creating a small API that uses
+[Deno KV](https://deno.com/kv) to store users in a database.
+
+Our project structure will look like this (in addition to the rest of the Fresh
+code from a new project):
+
+```
+├── routes
+│   └── api
+│       └── users
+│           ├── [id].ts
+│           └── index.ts
+```
+
+In each section about a method, only the relevant handler will be shown. The
+full files are available at the bottom for reference.
+
+## POST
+
+`POST` (create) is used to create a resource.
+
+```tsx
+// /routes/api/index.ts
+async POST(req, _ctx) {
+  const body = await req.text();
+  const user = JSON.parse(body) as User;
+  await upsertUser(user);
+  return new Response(JSON.stringify(user));
+},
```

Comment:
1. Let's display the full handler object here to allow users to copy & paste it. The snippet right now is invalid JS.

2. Let's use the [`request.json()`-method](https://developer.mozilla.org/en-US/docs/Web/API/Request/json) instead of parsing to JSON ourselves.

3. We parse the user as JSON only to serialize it back to a string again when we send it.

4. Let's inline the `upsertUser` function here. That way we see directly what's going on.

Reply:
1. I'll assume you mean the trimmed version just concerning the `POST` method, since the full object might have other methods as well. (In my test project, for example, I have `GET` implemented at the index as well, in order to get all users.) So something like this:
```ts
export const handler: Handlers<User | null> = {
  async POST(req, _ctx) {
    const user = await req.json() as User;
    await upsertUser(user);
    return new Response(JSON.stringify(user));
  },
};
```
2. Yes, that's what I wanted! Thanks.
3. I thought we were doing it to store the user. How else should I respond here?
4. Ok, and I'll try to clean this function up a bit.